vendor/tests.py

```diff
@@ -0,0 +1,209 @@
+from django.db import IntegrityError
+from django.test import TestCase
+
+# Create your tests here.
+
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from authentication.models import AppUser
+from .models import Vendor, PhotoVendor
+from .serializers import VendorSerializer, PhotoVendorSerializer
+
+class BaseTestCase(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = AppUser.objects.create_user(email='email@email.com',username='testuser',password='test')
+        self.client.force_authenticate(user=self.user)
+        self.vendor_data = {
+            "name": "Test Vendor",
+            "address": "123 Test St",
+            "price": 50,
+            "status": "PENDING",
+            "contact_name": "John Doe",
+            "contact_phone_number": "123-456-7890",
+            "event": 1,
+        }
+        self.vendor = Vendor.objects.create(**self.vendor_data)
+
+        self.event_id = 1
+
+        self.vendor1_data = {
+            "name": "Vendor 1",
+            "address": "123 Test St",
+            "price": 50,
+            "status": "PENDING",
+            "contact_name": "John Doe",
+            "contact_phone_number": "123-456-7890",
+            "event": self.event_id,
+        }
+        self.vendor1 = Vendor.objects.create(**self.vendor1_data)
+
+        self.photo = PhotoVendor.objects.create(
+            vendor=self.vendor,
+            image="https://example.com/path/to/your/image.jpg"
+        )
+
+        self.photo_data = {
+            "vendor": self.vendor.id,
+            "image": "https://example.com/path/to/your/image.jpg"
+        }
```

Comment:
There's some duplication in test data creation, particularly for vendors and photos. Consider refactoring this setup code into helper methods or decorators to reduce redundancy and improve readability. Overall the code is good and covers negative and positive test cases

Reply:
These two serves different purpose. `self.photo` is used to create an object that will be utilized later for output comparisons. On the other hand, `self.photo_data` is employed as raw data for the POST method. So, I feel like there is no actual duplication and there is no need to change the current code.